### PR TITLE
Christoffer, s205449

### PR DESCRIPTION
**Pull request by christofferfink**
Test setup and ci, can do both npm run test and npm test:watch - depending on use case.

Also setup workflow - might need review

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - main
+      - dev
+  pull_request:
+    branches:
+      - main
+      - dev
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
```
Comment (muItilingualism) [inline .github/workflows/ci.yml:15]: use ubuntu-22.04 like the backend CIs

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        uses: actions/setup-node@v2
+        with:
+          node-version: '22'
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Run tests
+        run: npm run test
+
```
Comment (muItilingualism) [inline .github/workflows/ci.yml:31]: Test before build? Shouldn't it be the other way around?

**Review by muItilingualism — CHANGES_REQUESTED** [review]
Looks pretty good, only looked at code tho

File: .github/workflows/ci.yml (hunk 2 above)
Comment (christofferfink) [reply .github/workflows/ci.yml:31]: Yea, you're right chatGPT said test first was best from a devops perspective 

But further research suggest your suggestion:
https://www.copado.com/platform-overview?utm_source=google&utm_medium=cpc&utm_campaign=google_search_emea_en_core_cicd_nonbrand&utm_content=DevOps&utm_term=devops%20platform&gad_source=1&gbraid=0AAAAACvgvepIm88s3fMg2cNLbBORHmyxi&gclid=Cj0KCQiA_qG5BhDTARIsAA0UHSL_7ZNZe_AYQD8UnQvjt-6nnUAH5Nn1zeUU9hdprgrxCJImHrSxhxcaAuvnEALw_wcB

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -26,8 +26,10 @@ jobs:
       - name: Install dependencies
         run: npm install
```
Comment (muItilingualism) [inline .github/workflows/ci.yml:27]: use `npm ci` instead of `npm install` (as `npm install` may mutate the package lock file, but `npm ci` does not)

**Review by muItilingualism — APPROVED** [review]

**Pull request merged**